util/a2e/atoe_utils.c

```diff
@@ -545,22 +548,29 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
+					char *tempPattern;
 
-					b = a2e((char *)pattern, fmt - pattern);
+					tempPattern = a2e((char *)pattern, fmt - pattern);
 
 					/* Extract a double from args, this works for both doubles
 					 * and floats,
 					 * NB if we use float for a single precision floating
 					 * point number the result is wrong.
 					 */
-					len = sprintf(this.buffer, b, va_arg(args, double));
-					free(b);
-					b = e2a_string(this.buffer);
-					strcpy(this.buffer, b);
-					free(b);
-					this.buffer += len;
+					size_t freeCap = count > this.counter ? count - this.counter : 0;
+					double argument = va_arg(args, double);
+					int requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
+					free(tempPattern);
+					CheckRet(requiredSpace);
+					this.counter += requiredSpace;
+					if(freeCap && requiredSpace)
```

Comment:
Space before `(`. Please pull the `{` from next line up to the end of this one too.

Reply:
I think we want to make the comparison explicit here.  `if ((freeCap > 0) && (requiredSpace > 0)) {`

Reply:
Thanks, done!